Makefile

```diff
@@ -203,38 +216,46 @@ uarch-with-toolchain:
+	cd src && $(INSTALL) tests/test-merkle-tree-hash tests/test-machine-c-api $(BIN_INSTALL_PATH)
+	cat tools/template/cartesi-machine-tests.template | sed 's|ARG_LUA_PATH|$(LUA_RUNTIME_PATH)/?.lua|g;s|ARG_LUA_CPATH|${LUA_DEFAULT_C_PATHS}|g;s|ARG_BIN_RUNTIME_PATH|${BIN_RUNTIME_PATH}|g' > $(BIN_INSTALL_PATH)/cartesi-machine-tests
+	cat tools/template/uarch-riscv-tests.template | sed 's|ARG_LUA_PATH|$(LUA_RUNTIME_PATH)/?.lua|g;s|ARG_LUA_CPATH|${LUA_DEFAULT_C_PATHS}|g;s|ARG_BIN_RUNTIME_PATH|${BIN_RUNTIME_PATH}|g' > $(BIN_INSTALL_PATH)/uarch-riscv-tests
+	cd $(BIN_INSTALL_PATH) && $(CHMOD_DATA) $(EMU_LUA_TEST_TO_BIN)
+	cd $(BIN_INSTALL_PATH) && $(CHMOD_EXEC) cartesi-machine-tests
+	patchelf --set-rpath $(LIB_RUNTIME_PATH) src/tests/test-merkle-tree-hash
+	patchelf --set-rpath $(LIB_RUNTIME_PATH) src/tests/test-machine-c-api
+
 install-emulator: $(BIN_INSTALL_PATH) $(LUA_INSTALL_CPATH)/cartesi $(LUA_INSTALL_PATH)/cartesi $(INC_INSTALL_PATH) $(IMAGES_INSTALL_PATH)
 	cd src && $(INSTALL) $(EMU_TO_BIN) $(BIN_INSTALL_PATH)
 	cd src && $(INSTALL) $(EMU_TO_LIB) $(LIB_INSTALL_PATH)
 	cd src && $(INSTALL) $(EMU_LUA_TO_BIN) $(BIN_INSTALL_PATH)
 	cd src && $(INSTALL) $(EMU_TO_LUA_CPATH) $(LUA_INSTALL_CPATH)
 	cd src && $(INSTALL) $(EMU_TO_LUA_CARTESI_CPATH) $(LUA_INSTALL_CPATH)/cartesi
 	cd src && $(INSTALL) $(EMU_TO_LUA_PATH) $(LUA_INSTALL_PATH)/cartesi
-	cat tools/template/cartesi-machine.template | sed 's|ARG_LUA_PATH|${LUA_DEFAULT_PATHS}|g;s|ARG_LUA_CPATH|${LUA_DEFAULT_C_PATHS}|g;s|ARG_INSTALL_PATH|${IMAGES_INSTALL_PATH}|g;s|ARG_BIN_INSTALL_PATH|${BIN_INSTALL_PATH}|g' > $(BIN_INSTALL_PATH)/cartesi-machine
-	cat tools/template/cartesi-machine-tests.template | sed 's|ARG_LUA_PATH|${LUA_DEFAULT_PATHS}|g;s|ARG_LUA_CPATH|${LUA_DEFAULT_C_PATHS}|g;s|ARG_BIN_INSTALL_PATH|${BIN_INSTALL_PATH}|g' > $(BIN_INSTALL_PATH)/cartesi-machine-tests
-	cat tools/template/cartesi-machine-stored-hash.template | sed 's|ARG_LUA_PATH|${LUA_DEFAULT_PATHS}|g;s|ARG_LUA_CPATH|${LUA_DEFAULT_C_PATHS}|g;s|ARG_BIN_INSTALL_PATH|${BIN_INSTALL_PATH}|g' > $(BIN_INSTALL_PATH)/cartesi-machine-stored-hash
-	cat tools/template/rollup-memory-range.template | sed 's|ARG_LUA_PATH|${LUA_DEFAULT_PATHS}|g;s|ARG_LUA_CPATH|${LUA_DEFAULT_C_PATHS}|g;s|ARG_BIN_INSTALL_PATH|${BIN_INSTALL_PATH}|g' > $(BIN_INSTALL_PATH)/rollup-memory-range
-	cat tools/template/uarch-riscv-tests.template | sed 's|ARG_LUA_PATH|${LUA_DEFAULT_PATHS}|g;s|ARG_LUA_CPATH|${LUA_DEFAULT_C_PATHS}|g;s|ARG_BIN_INSTALL_PATH|${BIN_INSTALL_PATH}|g' > $(BIN_INSTALL_PATH)/uarch-riscv-tests
-	cd $(BIN_INSTALL_PATH) && $(CHMOD_EXEC) $(EMU_TO_BIN) cartesi-machine cartesi-machine-tests cartesi-machine-stored-hash rollup-memory-range
+	cat tools/template/cartesi-machine.template | sed 's|ARG_LUA_PATH|$(LUA_RUNTIME_PATH)/?.lua|g;s|ARG_LUA_CPATH|$(LUA_DEFAULT_C_PATHS)|g;s|ARG_INSTALL_PATH|$(IMAGES_RUNTIME_PATH)|g;s|ARG_BIN_RUNTIME_PATH|$(BIN_RUNTIME_PATH)|g' > $(BIN_INSTALL_PATH)/cartesi-machine
```

Comment:
I packaged as the following:

```
make deb DESTDIR=$PWD/_install PREFIX=/usr/x86_64 MACHINE_EMULATOR_VERSION=0.14.0-dev
```

And this is the contents of `/usr/x86_64/bin/cartesi-machine`:
```
#!/bin/sh
export LUA_PATH="$(lua5.3 -e 'print(package.path)');/usr/x86_64/share/lua/5.3/?.lua"
export LUA_CPATH="$(lua5.3 -e 'print(package.cpath)');./?.so;/usr/local/lib/lua/5.3/?.so;/usr/local/share/lua/5.3/?.so;/opt/cartesi/lib/lua/5.3/?.so"
export CARTESI_IMAGES_PATH="/usr/x86_64/share/cartesi-machine/images"
lua5.3 "/usr/x86_64/bin/cartesi-machine.lua" "$@"
```

`LUA_PATH` is correct, but `LUA_CPATH` does not include the installed files in `/usr/x86_64/` and still references `/opt/cartesi/`. 

I think `LUA_DEFAULT_PATHS` and `LUA_DEFAULT_C_PATHS` are obsolete, and could be removed from the makefile, instead make use of `LUA_RUNTIME_CPATH/?.so`.

Reply:
Updated. running: `rm -rf _install/; make deb DESTDIR=$PWD/_install PREFIX=/usr/a/b/c` gave:

```
cat /usr/a/b/c/bin/cartesi-machine
#!/bin/sh
export LUA_PATH="$(lua5.3 -e 'print(package.path)');/usr/a/b/c/share/lua/5.3/?.lua"
export LUA_CPATH="$(lua5.3 -e 'print(package.cpath)');/usr/a/b/c/lib/lua/5.3/?.so"
export CARTESI_IMAGES_PATH="/usr/a/b/c/share/cartesi-machine/images"
lua5.3 "/usr/a/b/c/bin/cartesi-machine.lua" "$@"
```

And runs as:
```
/usr/a/b/c/bin/cartesi-machine
lua5.3: /usr/a/b/c/bin/cartesi-machine.lua:1025: unable to find length of flash drive 'root' image file '/usr/a/b/c/share/cartesi-machine/images/rootfs.ext2'
stack traceback:
        [C]: in function 'assert'
        /usr/a/b/c/bin/cartesi-machine.lua:1025: in local 'resolve_flash_lengths'
        /usr/a/b/c/bin/cartesi-machine.lua:1130: in main chunk
        [C]: in ?
```

Reply:
But this error is because this package does not install `rootfs.ext2` right? If you place it at `/usr/a/b/c/share/cartesi-machine/images/` should work I think.

Reply:
That was to show `cartesi-machine` running as expected.
Not to indicate a problem.
I looks misleading in retrospect.